### DISCUSSION TITLE
Invalid reference date format, try specifying one with the argument 'reference_format'

### DISCUSSION THREAD

**Discussion by ghost** [SilentVoid13/Templater#780] [Help]
**Plugin information (please complete the following information):**

-   OS: MacOS
-   Templater version: 1.12.0
-   Obsidian version: 0.14.2
-   Templater settings: 
![CleanShot 2022-04-06 at 20 32 04@2x](https://user-images.githubusercontent.com/61721378/161986444-9f95e67a-9ef7-4fcf-8aa5-f032a285ad62.jpg)

**Describe the bug**
Recently, everything work fine. But, today, when i create a new note with folder templates, i got this error: "Invalid reference date format, try specifying one with the argument 'reference_format'". Like image below:
![CleanShot 2022-04-06 at 20 28 17@2x](https://user-images.githubusercontent.com/61721378/161986806-73a095ea-6a3f-430f-87bc-39798bf16fe9.jpg)

This is my template file
![CleanShot 2022-04-06 at 20 28 02@2x](https://user-images.githubusercontent.com/61721378/161986902-9b8db7f0-26e6-4e31-b155-07db19eda7df.jpg)

Can someone help me? Thank you so much

**Comment by kenanmike** [comment]
@chumido98 which MacOS version are you on? 
I've got the same bug as well, and I saw question  from someone else on the Discord who also has it. 
The weird thing is I've been using these daily/weekly/monthly/quarterly/yearly note templates for months now without changing them and it suddenly broke. I updated to Monterey 12.3.1 not long ago, and updated my templater to the latest version (the one released in March). So maybe it is related to one of those two things? @SilentVoid13 any ideas?

**Comment by dredhorse** [comment]
is your note named with YYYY-MM-DD ?

**Comment by ghost** [comment]
> @chumido98 which MacOS version are you on? I've got the same bug as well, and I saw question from someone else on the Discord who also has it. The weird thing is I've been using these daily/weekly/monthly/quarterly/yearly note templates for months now without changing them and it suddenly broke. I updated to Monterey 12.3.1 not long ago, and updated my templater to the latest version (the one released in March). So maybe it is related to one of those two things? @SilentVoid13 any ideas?

Yes, i'm on Monterey

**Comment by ghost** [comment]
> is your note named with YYYY-MM-DD ?

Yes

**Comment by dredhorse** [comment]
ok, then try one line at a time. I had the same issue an in my case the note name wasn't a date as tp.file.title expected

**Comment by AB1908** [comment]
Y'all need to show us the templater command you're using. Hard to help without that haha

**Comment by motusRP** [comment]
Adding to this as I observed the same issue today while trying to diagnose something only slightly related.

I use a mostly stock Obsidian install, except I have a few of the Community plugins installed.  I don't think any settings would be useful but I can provide if someone would like to see them.

I have a dailyNotesTemplate.md file in my Templates/ folder that I use with Daily Notes.  It has a header that looks a lot like this (copied from the original post):
![image](https://user-images.githubusercontent.com/54725635/194143309-8fdee08b-560d-4926-bc1f-7e343f981808.png)

I tried applying my template to a manually created note, which by default had the name "Untitled".  It failed, with the same error as in the original post.  After diagnosing with the help of someone on Discord, I determined the issue was the default filename, which wasn't in the format YYYY-MM-DD.  Somewhere during the process of diagnosing this, however, I tried creating a second new note, which came up with the name "Untitled-2" by default.  When I tried applying the same template, *it worked*.  I believe this is unexpected behaviour, but probably not an "issue," so I didn't create a new issue for it.

I noticed that Untitled-2 and YYYY-MM-DD have the same number of characters... Could the check be just looking at string length and not the format of the content?

**Comment by dredhorse** [comment]
<% tp.date.now("YYYY-MM-DD",0, tp.file.title, "YYYY-MM-DD") %> 
does the following
- create a date = tp.date.now()
- with the format "YYYY-MM-DD" = 2022-10-05
- based on a 0 day offset
- froim a date which comes from tp.file.title
- and the title is formatted in "YYYY-MM-DD"

So if the filename is "Untitled" it doesn't work and will throw the error, it will also  not work for any title which is not in the format of "YYYY-MM-DD". So as an example if your title is "DP-2022-10-05" your second format of that line needs to be "[DP-]YYYY-MM-DD" to extract the right "Date" out of it.

Which information did you get in aliases with "Untitled-2" ??

From the code side Templater is using window.moment (which is the Moment library implemented in Obsidian if I'm correct) https://github.com/SilentVoid13/Templater/blob/2bd12daec648a5f97855d49fe112ec2a62579178/src/core/functions/internal_functions/date/InternalModuleDate.ts#L31

I doubt that they will only check the length of the string but you never know.
